arviz/data/io_numpyro.py

```diff
@@ -181,20 +181,26 @@ def sample_stats_to_xarray(self):
     @requires("posterior")
     @requires("model")
     def log_likelihood_to_xarray(self):
-        """Extract log likelihood from NumPyro posterior."""
+        """Extract log likelihood from NumPyro posterior using vectorization."""
         if not self.log_likelihood:
             return None
+
         data = {}
         if self.observations is not None:
             samples = self.posterior.get_samples(group_by_chain=False)
             if hasattr(samples, "_asdict"):
                 samples = samples._asdict()
-            log_likelihood_dict = self.numpyro.infer.log_likelihood(
-                self.model, samples, *self._args, **self._kwargs
-            )
+
+            # Vectorized log likelihood calculation using jax.vmap
+            log_likelihood_dict = jax.vmap(lambda single_sample: 
+                self.numpyro.infer.log_likelihood(self.model, single_sample, *self._args, **self._kwargs)
```

Comment:
```suggestion
                self.numpyro.infer.log_likelihood(self.model, single_sample, *self._args, batch_ndims=0, **self._kwargs)
```

It doesn't work without this because batching is not taken care of directly in vmap but this function expects a batch dimension too and fails when it is not there (or seemingly changes with the different variables)